README.md

```diff
@@ -162,6 +180,52 @@ comment:
     dot: true
 ```
 
+#### `comment.snippets[].exec`
+
+A path to a JavaScript file defining a function called `exec`.
+
+**Required**: false
+
+It can be used as a way to provide additional variables to the comment snippet body template, or to exclude the comment snippet from the comment even when the globs matched some of the files changed in the PR. 
+
+This function will receive as arguments:
+
+- `changedFiles` - a list of file paths that matched this comment snippet's globs (`comment.snippets[].files`).
+- `pr` - the whole response of a [`GET /repos/{owner}/{repo}/pulls/{pull_number}` request](https://docs.github.com/en/rest/reference/pulls#get-a-pull-request).
+
+Note that values such as `pr.body`, `pr.title`, or `pr.head.ref` should be considered [unsafe user input](https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#understanding-the-risk-of-script-injections).
+
+It should return one of:
+- `false` if the comment snippet should be excluded from the comment.
+- `true` if the comment snippet should be included included in the comment.
+- an object with additional variables for the comment body template if needed.
+
+The script runs in a limited context, e.g. you cannot call `require` or `process`.
+
+`fs.readFileSync` is available as `readFileSync` so that you can inspect the current contents of all the files in the repository.
```

Comment:
Those comments make sense assuming that https://nodejs.org/dist/latest-v14.x/docs/api/vm.html#vm_vm_runincontext_code_contextifiedobject_options will be used to isolate the function. I'm not yet 100% sure if that's the best and safest way.

I was considering spawning a new process and make it communicate with the parent by standard output but I cannot really control if the `exec` function will litter stdout with some `console.log`s or not.